ip/event_manager.go

```diff
@@ -164,15 +164,22 @@ func (m *manager) singleEtcdRun(ctx context.Context) {
 	log := logger.Get(ctx)
 	err := m.locker.Refresh(ctx)
 	if err != nil {
-		log.WithError(err).Error("Fail to refresh lock")
-		m.sendEvent(FaultEvent)
+		m.keepaliveRetry++
+		log.WithError(err).Info("Fail to refresh lock (retry)")
+		if m.keepaliveRetry > m.config.KeepAliveRetry {
+			log.WithError(err).Error("Fail to refresh lock")
+			m.sendEvent(FaultEvent)
+		}
 		return
 	}
+	m.keepaliveRetry = 0
 
 	isMaster, err := m.locker.IsMaster(ctx)
 	if err != nil {
 		log.WithError(err).Error("Fail to check lock")
-		m.sendEvent(FaultEvent)
+		// Fault event should only be handled in the Refresh operation where the
+		// retry loop is present
```

Comment:
I don't undestand this. Why ?

Reply:
I've decided this to simply the retry logics, actually both are related to handling the lock, the first one is about acquiring it, the second is just checking its state. I think that we can skip this error, and keep the retry things in the Refresh part which is the important one?

Reply:
If you tell me it's a bad design and something is wrong with this I can change it but it seemed good to me.